app/models/box_form.rb

```diff
@@ -185,4 +197,25 @@ def have_virus_sample?
   def have_distractor_sample?
     @samples.any? { |_, sample| sample.distractor }
   end
+
+  def initialize_csv_box
+    CSV.open(@csv_box.path) do |csv_stream|
+      i = 0
+      csv_stream.each do |row|
+        batch_number, concentration, distractor, instruction = row[0..3]
+        batch_uuid = Batch.find_by(batch_number: batch_number)&.uuid
+        next if batch_uuid.blank?
+        @batches_data[i] = {
+          batch_uuid: batch_uuid,
+          distractor: distractor.downcase == "yes",
+          instruction: instruction,
+          concentrations: {i: {
+            replicate: 1,
+            concentration: Integer(Float(concentration)),
+          }},
```

Comment:
Note that you're actually creating a `{ :i => {} }` hash (symbol), when I think you mean `{ i => {} }`, but actually you could just go with `{ "0" => {} }`, since there will always be only one entry, unless you rework to group by batch UUID, which might be a good idea.